cloud/openstack/os_keystone_role.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/python
+# Copyright (c) 2016 IBM
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+DOCUMENTATION = '''
+---
+module: os_keystone_role
+short_description: Manage OpenStack Identity Roles
+extends_documentation_fragment: openstack
+version_added: "2.1"
+author: "Monty Taylor (@emonty), David Shrewsbury (@Shrews)"
+description:
+    - Manage OpenStack Identity Roles.
+options:
+   name:
+     description:
+        - Role Name
+     required: true
+   state:
+     description:
+       - Should the resource be present or absent.
+     choices: [present, absent]
+     default: present
+requirements:
+    - "python >= 2.6"
+    - "shade"
+'''
+
+EXAMPLES = '''
+# Create a role named "demo"
+- os_keystone_role:
+    cloud: mycloud
+    state: present
+    name: demo
+
+# Delete the role named "demo"
+- os_keystone_role:
+    cloud: mycloud
+    state: absent
+    name: demo
+'''
+
+RETURN = '''
+role:
+    description: Dictionary describing the role.
+    returned: On success when I(state) is 'present'.
+    type: dictionary
+    contains:
+        id:
+            description: Unique role ID.
+            type: string
+            sample: "677bfab34c844a01b88a217aa12ec4c2"
+        name:
+            description: Role name.
+            type: string
+            sample: "demo"
```

Comment:
I'm getting back `HUMAN_ID` and `NAME_ATTR` and `human_id` from my cloud, as well as a `links` hash.

```
<localhost> REMOTE_MODULE os_keystone_role cloud=trustytest name=jlkderp
localhost | success >> {
    "changed": false,
    "role": {
        "HUMAN_ID": false,
        "NAME_ATTR": "name",
        "human_id": null,
        "id": "27d4a926fa5a4d69ab0d3709ebaf204a",
        "links": {
            "self": "https://trustytest.openstack.blueboxgrid.com:5001/v3/roles/27d4a926fa5a4d69ab0d3709ebaf204a"
        },
        "name": "jlkderp"
    }
}
```

Not different enough to block this PR, just informational data.